src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/AssessmentService.kt

```diff
@@ -235,6 +235,32 @@ class AssessmentService(
     return AuthorisableActionResult.Success(clarificationNoteEntity)
   }
 
+  fun updateAssessmentClarificationNote(user: UserEntity, assessmentId: UUID, id: UUID, response: String): AuthorisableActionResult<ValidatableActionResult<AssessmentClarificationNoteEntity>> {
+    val assessment = when (val assessmentResult = getAssessmentForUser(user, assessmentId)) {
+      is AuthorisableActionResult.Success -> assessmentResult.entity
+      is AuthorisableActionResult.Unauthorised -> return AuthorisableActionResult.Unauthorised()
+      is AuthorisableActionResult.NotFound -> return AuthorisableActionResult.NotFound()
+    }
+
+    var clarificationNoteEntity = assessmentClarificationNoteRepository.findByAssessmentIdAndId(assessment.id, id)
+
+    if (clarificationNoteEntity === null) {
+      return AuthorisableActionResult.NotFound()
+    }
+
+    if (clarificationNoteEntity.createdByUser !== user) {
```

Comment:
This is a bit of a subtle one if you've still got your JS/TS hat on - in Kotlin `==` means `use the equals method on the class to compare two instances` and `===` means `check if these are the same object/location in memory.`  Because `UserEntity` is a data class it gets an equals method generated by the compiler that compares the values of its properties - so we probably want to use that here in case we end up with more two different objects representing the same user entry in the DB.

For more info: https://kotlinlang.org/docs/equality.html